portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -63,6 +63,16 @@ public static Comment fromEntity(Entity entity) {
     return new Comment(text, name, entity.getKey(), time);
   }
 
+  /** Returns true if text or name (inclusive) contains the filter string, ignoring case */
+  public boolean contains(String filter) {
+    if(filter == null) {
+      return true;
+    }
+    // Ignore case
+    filter = filter.toLowerCase();
```

Comment:
It's important to consider costs of copying data.

Here's a sample analysis. 
Filter(comments) => calls this method N times, once for every comment. That means that filter.toLowerCase() will be called N times, causing N full copies of the filter. in reality, you only need one copy of the filter. 

Instead, consider using Consider instead using Apache Commons containsIgnoreCase().

https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#containsIgnoreCase-java.lang.CharSequence-java.lang.CharSequence-

This implementation doesn't do copying, as far as I can tell, and would therefore perform much faster in practice.

Reply:
Additionally, this library happens to handle the "null" case for you, further reducing the lines of code you need to write :).